### PR TITLE
bt-25: Add textarea component

### LINKED ISSUE #25: Create TextArea component
Image

https://mui.com/base-ui/react-textarea-autosize/

### PR DESCRIPTION
**Pull request by lavandosya**
I added textarea component

Screenshot 2023-08-21 at 19 40 21

The example of usage:

Screenshot 2023-08-21 at 19 41 14

### CODE REVIEW

File: frontend/src/bundles/common/components/textarea/styles.module.scss
Comment (nikita-remeslov) [inline frontend/src/bundles/common/components/textarea/styles.module.scss:1]: not sure about styles. I think we need to create some variables for it
Let's discuss this on daily

File: frontend/src/bundles/common/components/textarea/textarea.tsx [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+    maxRows,
+    placeholder = '',
+}: Properties<T>): JSX.Element => {
+    const { field } = useFormController({ name, control });
+
+    return (
+        <div className={styles.container}>
+            <label className={styles.label} htmlFor={field.name}>
+                {label}
+            </label>
```
Comment (nikita-remeslov) [inline frontend/src/bundles/common/components/textarea/textarea.tsx]: we will have separate component for label

**Review by nikita-remeslov — CHANGES_REQUESTED** [review]

File: frontend/src/assets/css/variables.css [hunk 2]
```diff
@@ -0,0 +1,5 @@
+:root {
+  --text-light: #90a3bf;
```
Comment (nikita-remeslov) [inline frontend/src/assets/css/variables.css]: Looks like it our design colors.
image

We should add proper name for it.
Example: `bt-gray`
or use some tools for it - https://www.color-name.com/hex/90a3bf

File: frontend/src/assets/css/variables.css [hunk 3]
```diff
@@ -0,0 +1,5 @@
...
+  --hover: #0084dd;
+  --input: #d5dce8;
```
Comment (nikita-remeslov) [inline frontend/src/assets/css/variables.css]:
```suggestion
  --bt-input: #d5dce8;
```

File: frontend/src/assets/css/variables.css [hunk 4]
```diff
@@ -0,0 +1,5 @@
+:root {
+  --text-light: #90a3bf;
+  --hover: #0084dd;
```
Comment (nikita-remeslov) [inline frontend/src/assets/css/variables.css]: image
Looks like for focused/hover state we have different collor

File: frontend/src/bundles/common/components/textarea/textarea.tsx [hunk 5]
```diff
@@ -0,0 +1,41 @@
...
+    maxRows: number;
+    placeholder?: string;
+};
+
+const Textarea = <T extends FieldValues>({
+    control,
+    name,
+    minRows,
+    maxRows,
+    placeholder = '',
```
Comment (nikita-remeslov) [inline frontend/src/bundles/common/components/textarea/textarea.tsx]:
```suggestion
    control,
    name,
    placeholder = '',
    ...props
```

File: frontend/src/bundles/common/components/textarea/textarea.tsx [hunk 6]
```diff
@@ -0,0 +1,41 @@
...
+}: Properties<T>): JSX.Element => {
+    const { field } = useFormController({ name, control });
+
+    return (
+        <TextareaAutosize
+            {...field}
+            minRows={minRows}
+            maxRows={maxRows}
+            className={styles.textarea}
+            placeholder={placeholder}
```
Comment (nikita-remeslov) [inline frontend/src/bundles/common/components/textarea/textarea.tsx]:
```suggestion
            className={styles.textarea}
            placeholder={placeholder}
            {...props}
```

File: frontend/src/bundles/common/components/textarea/textarea.tsx (hunk 5 above)
Comment (nikita-remeslov) [reply frontend/src/bundles/common/components/textarea/textarea.tsx]: we can simplify

File: frontend/src/bundles/common/components/textarea/textarea.tsx (hunk 6 above)
Comment (nikita-remeslov) [reply frontend/src/bundles/common/components/textarea/textarea.tsx]: we can simplify

**Review by nikita-remeslov — CHANGES_REQUESTED** [review]

File: frontend/src/assets/css/variables.css (hunk 2 above)
Comment (lavandosya) [reply frontend/src/assets/css/variables.css]: okay, thanks

**Comment by lavandosya** [comment]
I've done change request, could you please review it?

**Review by anton-otroshchenko — APPROVED** [review]

File: frontend/src/assets/css/variables.css [hunk 7]
```diff
@@ -0,0 +1,5 @@
+:root {
```
Comment (nikita-remeslov) [inline frontend/src/assets/css/variables.css:1]: Maybe we need to update the name for these colors. But now we can keep it

**Review by nikita-remeslov — APPROVED** [review]

**Pull request merged**